### PR TITLE
Constructing PageAccessor and using it to create _fingerprints

### LINKED ISSUE #192: Clean allocations in Xor
#191 introduced great performance gains but added a bit of allocs. It can be removed and pool can be leveraged more. The list of allocations:

- [x] `fp` - in #259 
- [ ] `_fingerprints` (see: [code](https://github.com/NethermindEth/Paprika/blob/cda38d382fd823b4e2dacde6573db344f61d2af5/src/Paprika/Utils/Xor8.cs#L28)) and replacing them with something based on the `BufferPool`

The last one requires constructing a page accessor that would allow to use multiple pages/buffers from the pool and dispose them properly at the end. It could something as simple as `Page[]` as it would already provide a good fanout 1 item -> 4kb of memory.

### PR DESCRIPTION
**Pull request by shubham-sonthalia**
Resolves #192

### CODE REVIEW

**Review by Scooletz — CHANGES_REQUESTED** [review]
This solution provides a managed solution that we want to go aware from. There's the `BufferPool` already that provides `4kb` chunks that are used by `PooledSpanDictionary` and others. Would it be possible to have it migrated to use this option? Otherwise, these chunks cannot be shared and we land in a world where different parts of Paprika used different pools.

File: src/Paprika/Utils/Xor8.cs [hunk 1]
```diff
@@ -216,6 +221,10 @@ private int GetHash(ulong key, ulong seed, int index)
 
     private static int Fingerprint(ulong hash) => (int)(hash & ((1 << BitsPerFingerprint) - 1));
 
+    ~Xor8()
```
Comment (benaadams) [inline src/Paprika/Utils/Xor8.cs:224]: Finalizers are bad for performance, would be better to use IDisposable or similar

**Comment by shubham-sonthalia** [comment]
> This solution provides a managed solution that we want to go aware from. There's the `BufferPool` already that provides `4kb` chunks that are used by `PooledSpanDictionary` and others. Would it be possible to have it migrated to use this option? Otherwise, these chunks cannot be shared and we land in a world where different parts of Paprika used different pools.

Sure, will work on integrating the BufferPool instead of the new class.

File: src/Paprika/Utils/Xor8.cs (hunk 1 above)
Comment (shubham-sonthalia) [reply src/Paprika/Utils/Xor8.cs:224]: Yes, will remove it.

**Comment by Scooletz** [comment]
#345 replaces this providing an ORable filter that uses the unmanaged memory

**Pull request closed without merging**